### ISSUE TITLE
[Bug]: invalid value encountered with 'ortho' projection mode 

### ISSUE THREAD

**Issue by patquem** [matplotlib/matplotlib#22624]
### Bug summary

Hello,

Passing the mouse on the figure when plot has been generated with **'ortho'** projection 3d mode and specific azimuthal and elevation angles returns the following message : 

_C:\....\lib\site-packages\mpl_toolkits\mplot3d\proj3d.py:25: RuntimeWarning: invalid value encountered in double_scalars
  u = (x01*x21 + y01*y21) / (x21¨¨2 + y21¨¨2)_

Could you please investigate this ?
thanks,
Patrick

### Code for reproduction

```python
import matplotlib.pyplot as plt

fig, ax = plt.subplots(subplot_kw={'projection': '3d'})
ax.scatter(0, 0)
ax.set_proj_type('ortho')
ax.view_init(azim=-90, elev=90)
plt.show()
```

### Actual outcome

C:\....\lib\site-packages\mpl_toolkits\mplot3d\proj3d.py:25: RuntimeWarning: invalid value encountered in double_scalars
  u = (x01*x21 + y01*y21) / (x21**2 + y21**2)

> 

### Expected outcome

No RuntimeWarning message

### Operating system

Windows

### Matplotlib Version

3.4.2

### Python version

3.7.7

### Installation

pip

**Comment by oscargus** [comment]
The warning occurs the first time x21 and y21 both are 0 (so it is a divide by zero error).

A simple way fix it is to change the code to something like:
```
    if x21 != 0 or y21 != 0:
        u = (x01*x21 + y01*y21) / (x21**2 + y21**2)
        u = np.clip(u, 0, 1)
    else:
        u = 0   # Actually doesn't matter since it will be multiplied by zero anyway...
```
although the preferred syntax may differ.

**Comment by oscargus** [comment]
Seems like one should use `invalid='ignore'` as although there is a division by zero, the actual warning is "Invalid..." (probably because it is 0/0). For now, I went with my approach (as I didn't figure out the answer above until after pushing it...), but we'll see what the review leads to.

**Comment by anntzer** [comment]
The warning is triggered when trying to get xyz coordinates for the mouse cursor and one of the axes is exactly orthogonal to the screen, but I wonder whether printing xyz coordinates in the toolbar even makes sense at all?  Basically, we pick an arbitrary screen-z-depth for the cursor (based on the distance to the edges -- which is where the warning comes from) and print the corresponding xyz value, but if one e.g. carefully looks at the displayed xyz values while slowly moving the mouse across a (default, non-ortho) 3d axes, one will see that the value suddenly jumps when the mouse goes from "closer to one axis" to "closertto another axis".
In other words, I would suggest to just stop displaying mouse cursor xyz values here.

**Comment by WeatherGod** [comment]
Displaying the (estimated) 3D pointer coordinates has been a feature in
mplot3d for over 10 years. The code that is used to compute this estimate
has been reused in other projects as well, and when this feature has broken
for other reasons, users have noticed and complained. I see no reason to
get rid of it entirely.

Now, if a better algorithm for computing the estimated coordinates can be
devised, all the better!

**Comment by anntzer** [comment]
I am not convinced the values make sense at all.  For example, these are the xyz values printed when dragging the mouse across window (default 3d view) horizontally, along the middle y position:
```python
from pylab import *
rcdefaults(); rcParams["axes.unicode_minus"] = False  # use ascii minus
fig = figure()
ax = fig.add_subplot(projection="3d")
fig.canvas.draw()
xyzs = []
screen_y = .5
# drag mouse across window horizontally, along the middle y position
for screen_x in np.linspace(0, 1, 1001):
    # formatted string
    s = ax.format_coord(
        *ax.transData.inverted().transform(
            fig.transFigure.transform((screen_x, screen_y))))
    # displayed xyz
    xyzs.append([float(part.split("=")[1]) for part in s.split(",")])
xyzs = array(xyzs)
close(fig)
plot(xyzs[:, 0], label="x")
plot(xyzs[:, 1], label="y")
plot(xyzs[:, 2], label="z")
legend()
show()
```
![xyz](https://user-images.githubusercontent.com/1322974/157690564-211a99b4-7112-424f-99da-dfb1a3536423.png)

The farthest side branches can probably be ignored (the mouse is out of the axes), but the middle jumping is... not nice.

**Comment by anntzer** [comment]
> when this feature has broken for other reasons, users have noticed and complained

Actually, I have quickly skimmed through the issue list and didn't find anything related, but perhaps you have refs e.g. on the mailing list?

**Comment by WeatherGod** [comment]
Ok, so I am also having difficulty finding what I was thinking of, but, here are some issues where people expressed interest in having access to 3D coordinate information:

https://github.com/matplotlib/matplotlib/issues/16156
https://github.com/matplotlib/matplotlib/issues/11309
https://discourse.matplotlib.org/t/no-zdata-in-motion-notify-event-when-using-a-3d-plot/16201

This isn't a comprehensive listing of such interest, and it shows that users find the information that is displayed by `format_coords()` to be useful enough to want to know more about how to get it programmatically. While I haven't had time to look closely at what you showed above, that isn't an argument to get rid of it; it is an argument to make it better behaved.

**Comment by anntzer** [comment]
I don't disagree with making it better behaved.  However, the value currently displayed is simply worse than useless, it is misleading.
